spec/support/crypt.rb

```diff
@@ -0,0 +1,77 @@
+# Copyright (C) 2009-2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Crypt
+
+  # For tests that require local KMS to be configured
+  shared_context 'with local kms_providers' do
+    let(:kms_provider_name) { 'local' }
+    let(:kms_providers) { local_kms_providers }
+
+    let(:data_key) do
+      BSON::ExtJSON.parse(File.read('spec/mongo/crypt/data/key_document_local.json'))
+    end
+
+    let(:schema_map) do
+      BSON::ExtJSON.parse(File.read('spec/mongo/crypt/data/schema_map_local.json'))
+    end
+  end
+
+  # For tests that require AWS KMS to be configured
+  shared_context 'with AWS kms_providers' do
+    let(:kms_provider_name) { 'aws' }
+    let(:kms_providers) { aws_kms_providers }
+
+    let(:data_key) do
+      BSON::ExtJSON.parse(File.read('spec/mongo/crypt/data/key_document_aws.json'))
+    end
+
+    let(:schema_map) do
+      BSON::ExtJSON.parse(File.read('spec/mongo/crypt/data/schema_map_aws.json'))
+    end
+  end
+
+  def self.included(context)
```

Comment:
Can all of these methods be moved into a RSpec shared context, and then the Crypt module can be included globally in rspec config.